pydicom/filewriter.py

```diff
@@ -22,6 +22,92 @@
 from pydicom.tagtools import tag_in_exception
 
 
+def correct_ambiguous_vr_element(elem, ds, is_little_endian):
+    """Attempt to correct the ambiguous VR element `elem`.
+
+    When it's not possible to correct the VR, the element will be returned
+    unchanged. Currently the only ambiguous VR elements not corrected for are
+    all retired or part of DICONDE, except for (60xx,3000) Overlay Data.
+
+    If the VR is corrected and is 'US' or 'SS then the value will be updated
+    using the pydicom.values.convert_numbers() method.
+
+    Parameters
+    ----------
+    elem : pydicom.dataelem.DataElement
+        The element with an ambiguous VR.
+    ds : pydicom.dataset.Dataset
+        The dataset containing `elem`.
+    is_little_endian : bool
+        The byte ordering of the values in the dataset.
+
+    Returns
+    -------
+    ds : pydicom.dataset.Dataset
+        The corrected dataset
+    """
+    if 'or' in elem.VR:
+        # 'OB or OW': 7fe0,0010 PixelData
+        if elem.tag == 0x7fe00010:
+            # If BitsAllocated is > 8 then OW, else may be OB or OW
+            #   As per PS3.5 Annex A.2. For BitsAllocated < 8 test the size
+            #   of each pixel to see if its written in OW or OB
+            try:
+                if ds.BitsAllocated > 8:
```

Comment:
I raise these points because the `correct_ambiguous_vr_element` function is actually useful to me when writing a new dicom file / recompressing the pixel data. For example I have the following in my code:
```python
            ds.PixelData = pixel_data
            ds.file_meta.TransferSyntaxUID = JPEGBaseLineLossy8bit
            ds.is_little_endian = True
            # Compressed pixel data is always explicit VR
            ds.is_implicit_VR = False

            pixel_data_tag = ds[(0x7fe0, 0x0010)]
            pixel_data_tag.is_undefined_length = True
            if pixel_data_tag.VR == 'OB or OW':
                pixel_data_tag.VR = 'OB'
```